internal/pkg/mapper/sharedcode/links/save_test.go

```diff
@@ -13,53 +14,102 @@ func TestSharedCodeLinksMapBeforeLocalSave(t *testing.T) {
-	sharedCodeState := sharedCodeStateRaw.(*model.ConfigState)
-	sharedCodeState.SetLocalState(&model.Config{
-		ConfigKey: sharedCodeKey,
-		Content: utils.PairsToOrderedMap([]utils.Pair{
-			{Key: model.SharedCodeComponentIdContentKey, Value: `keboola.python-transformation-v2`},
-		}),
-	})
+	// Shared code config with rows
+	fixtures.CreateSharedCode(t, context.State, context.Naming)
 
 	// Config using shared code
 	configKey := model.ConfigKey{
 		BranchId:    123,
 		ComponentId: `keboola.python-transformation-v2`,
 		Id:          `789`,
 	}
-	config := &model.Config{
-		ConfigKey: configKey,
-	}
-	configManifest := &model.ConfigManifest{
-		ConfigKey: configKey,
+	configState := &model.ConfigState{
+		ConfigManifest: &model.ConfigManifest{
+			ConfigKey: configKey,
+		},
+		Local: &model.Config{
+			ConfigKey: configKey,
+			Content: utils.PairsToOrderedMap([]utils.Pair{
+				{
+					Key:   model.SharedCodeIdContentKey,
+					Value: `456`,
+				},
+				{
+					Key:   model.SharedCodeRowsIdContentKey,
+					Value: []string{`1234`, `5678`},
+				},
+			}),
+			Blocks: model.Blocks{
+				{
+					Name: `Block 1`,
+					Codes: model.Codes{
+						{
+							CodeKey: model.CodeKey{
+								ComponentId: `keboola.python-transformation-v2`,
+							},
+							Name: `Code 1`,
+							Scripts: []string{
+								`print(100)`,
+								" {{1234}}\n",
+							},
+						},
+						{
+							CodeKey: model.CodeKey{
+								ComponentId: `keboola.python-transformation-v2`,
+							},
+							Name: `Code 2`,
+							Scripts: []string{
+								" {{5678}}\n",
+								"{{1234}}",
+							},
+						},
+					},
+				},
+			},
+		},
 	}
-	recipe := createLocalSaveRecipe(config, configManifest)
-	recipe.Configuration.Content.Set(model.SharedCodeIdContentKey, `456`)
+	assert.NoError(t, context.State.Set(configState))
 
 	// Invoke
+	recipe := createLocalSaveRecipe(configState.Local, configState.ConfigManifest)
 	assert.NoError(t, mapperInst.MapBeforeLocalSave(recipe))
 	assert.Empty(t, logs.String())
 
 	// Path is replaced by ID
 	_, found := recipe.Configuration.Content.Get(model.SharedCodeIdContentKey)
 	assert.False(t, found)
+	_, found = recipe.Configuration.Content.Get(model.SharedCodeRowsIdContentKey)
+	assert.False(t, found)
 	sharedCodeId, found := recipe.Configuration.Content.Get(model.SharedCodePathContentKey)
 	assert.True(t, found)
 	assert.Equal(t, sharedCodeId, `_shared/keboola.python-transformation-v2`)
+
+	// IDs in transformation blocks are replaced by paths
+	assert.Equal(t, model.Blocks{
+		{
+			Name: `Block 1`,
+			Codes: model.Codes{
+				{
+					CodeKey: model.CodeKey{
+						ComponentId: `keboola.python-transformation-v2`,
+					},
+					Name: `Code 1`,
+					Scripts: []string{
+						`print(100)`,
+						"# {{:codes/code1}}",
+					},
+				},
+				{
+					CodeKey: model.CodeKey{
+						ComponentId: `keboola.python-transformation-v2`,
+					},
+					Name: `Code 2`,
+					Scripts: []string{
+						"# {{:codes/code2}}",
```

Comment:
Test id -> path.

Reply:
V lokalnej reprezentacii zacina placeholder volitelne s komentarom.
Je to preto aby bol subor, napr. `code.py` validny, a IDE nehlasilo chybu.